lodash-deep/lodash-deep.d.ts

```diff
@@ -0,0 +1,14 @@
+// Type definitions for lodash-deep v 2.0
```

Comment:
The name of this file should be `index.d.ts`. Also, the `lodash-deep` folder should be inside the `types/` folder.

Reply:
@armanio123, Thank you for all the help. I am new to this, can you send me some literature I can read so I can become better at it ?

Reply:
You can find most of the documentation on [How can I contribute](https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute) section of the DefinitelyTyped readme. 

Also the [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/basic-types.html) is a great source for understanding details about the types.

Reply:
@armanio123, Thank you and could you please re-review ?